### PR TITLE
Allow the Firefox Accounts environment to be parameterised

### PR DESCRIPTION
**Pull request by davehunt**
This patch allows anything that accepts a parameterised mark to override the environment. This means that if you have a single test that requires the production environment, you can just add the appropriate mark to that test. It also allows tests to be executed against multiple environments by providing additional parameters.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -63,8 +63,28 @@ Specifying an environment
...
 By default all accounts will be created on the 'stage' environment. You can set
 the ``FXA_ENV`` environment variable to target 'production' or 'stable'.
-Alternatively, you can override the ``fxa_urls`` fixture to specify the URL for
-you environment:
+
+If you need to override the environment for a subset of tests, or run tests against multiple environments, you can do this using parameterisation:
+
+.. code-block:: python
+
+  @pytest.mark.parametrize('fxa_urls', ['production'], indirect=True)
```
Comment (b4handjr) [inline README.rst]: This is nice!

**Comment by hackebrot** [comment]
Hi @davehunt @jrbenny35! 👋 

I was curious if there might be a good way to make the API for users a bit nicer, meaning that they don't have to know about the ``urls`` fixture and how ``indirect`` works. I came up with [this](https://gist.github.com/hackebrot/5879990ebb0603dacbbde83d7253ddd8).

**Comment by davehunt** [comment]
> I was curious if there might be a good way to make the API for users a bit nicer

My initial thought is that this is an unusual and somewhat advanced feature, so I'm not concerned about the API being a little convoluted.. but I do like how your solution looks for the user. I'll see what I can come up with using your snippet.

**Comment by davehunt** [comment]
@hackebrot @chartjes let me know what you think, cheers!

File: README.rst [hunk 2]
```diff
@@ -63,8 +63,28 @@ Specifying an environment
...
+  def test_production(fxa_account):
+      selenium.get('https://example.com/')
+      selenium.find_element(By.ID, 'email').send_keys(fxa_account.email)
+      selenium.find_element(By.ID, 'password').send_keys(fxa_account.password)
+      selenium.find_element(By.ID, 'login').click()
+
+
+  @pytest.mark.fxa_env('stage', 'stable')
+  def test_stage(fxa_account):
+      selenium.get('https://elpmaxe.com/')
```
Comment (hackebrot) [inline README.rst:81]: It doesn't really make a difference, but is this supposed to be ``https://example.com/``? 🤔

File: pytest_fxa/plugin.py [hunk 3]
```diff
@@ -28,14 +28,35 @@ def pytest_addoption(parser):
...
+    if 'fxa_urls' not in metafunc.fixturenames:
+        return
+
+    envs = set([
+        env_name
+        for marker in metafunc.definition.iter_markers('fxa_env')
+        for env_name in marker.args])
+
+    if envs:
+        metafunc.parametrize('fxa_urls', envs, indirect=True)
```
Comment (hackebrot) [inline pytest_fxa/plugin.py:48]: What happens if the ``fxa_env`` marker is applied without any arguments?

I assume the corresponding test would error, since ``fxa_urls`` doesn't receive a ``request`` object?! 🤔

File: tests/test_plugin.py [hunk 4]
```diff
@@ -90,3 +90,30 @@ def test_account(fxa_account):
...
+
+        def test_account(fxa_urls):
+            assert '{0}' in fxa_urls['authentication']
+    """.format(env))
+    result = testdir.runpytest()
+    result.assert_outcomes(passed=1)
+
+
+def test_fxa_env_marker(monkeypatch, testdir):
+    env, marker = ['stable', 'stage']
```
Comment (hackebrot) [inline tests/test_plugin.py:109]: It would be good to have an extra test that makes sure the marker can handle a sequence of environments.

**Review by hackebrot — COMMENTED** [review]
Looks good! Few questions around edge cases and testing.

File: README.rst (hunk 2 above)
Comment (davehunt) [reply README.rst:81]: That would be an impressive typo! Mozilla has a tendancy to register hostnames for staging environments that are transposed versions of the production hosts. For example mozilla.org becomes allizom.org... this is playfully doing the same with example.com.

File: pytest_fxa/plugin.py (hunk 3 above)
Comment (davehunt) [reply pytest_fxa/plugin.py:48]: No error, we default to staging. I'll add a test to make that clearer.

File: tests/test_plugin.py (hunk 4 above)
Comment (davehunt) [reply tests/test_plugin.py:109]: Agreed, coming up!

File: README.rst (hunk 2 above)
Comment (hackebrot) [reply README.rst:81]: Haha got it! 👍

File: tests/test_plugin.py (hunk 4 above)
Comment (hackebrot) [reply tests/test_plugin.py:109]: Great! 🚀

File: pytest_fxa/plugin.py (hunk 3 above)
Comment (hackebrot) [reply pytest_fxa/plugin.py:48]: Ahh got it! 👍

**Review by hackebrot — APPROVED** [review]
LGTM 👍
